langchain4j/src/main/java/dev/langchain4j/service/ServiceOutputParser.java

```diff
@@ -115,34 +115,36 @@ public static String outputFormatInstructions(Class<?> returnType) {
             return "\nYou must put every item on a separate line.";
         }
 
-        return "\nYou must answer strictly in the following JSON format: " + jsonStructure(returnType);
+        return "\nYou must answer strictly in the following JSON format: " + jsonStructure(returnType, new HashSet<>());
     }
 
-    private static String jsonStructure(Class<?> structured) {
+    private static String jsonStructure(Class<?> structured, Set<String> visited) {
```

Comment:
minor: would be probably safer and more clear to use `Set<Class<?>> visited` instead of `Set<String> visited`